src/test/java/com/puppycrawl/tools/checkstyle/PropertyCacheFileTest.java

```diff
@@ -328,7 +328,7 @@ public void testSymbolicLinkResolution() throws IOException {
     @DisabledOnOs(OS.WINDOWS)
     public void testSymbolicLinkToNonDirectory() throws IOException {
         final Path tempFile = Files.createTempFile("tempFile", null);
```

Comment:
@rnveach , in response to the concluding comment on this PR, i still haven't figured out a way to get rid of `createTempFile`. `@TempDir` annotation helped to replace `createTempDirectory`. But, i am afraid if there is any similar annotation like `@TempDir` as such for the `createTempFile` method. But, given my understanding of @TempDir, it will delete all the files of the contents on deletion. So, forbidding createTempFile, is it needed? Also, the issue mentioned of `Files.createTempDirectory`. I went on to add Files.createTempFile in the signature-test.txt as i had some review comments on the same.

Reply:
@rnveach , @relentless-pursuit ,
Firstly, sorry to involve.

> But, given my understanding of @TempDir, it will delete all the contents including temp files on completition of execution. So, forbidding Files.createTempFile, is it needed?

I don't think the api guarantees that the files will be cleaned up when the JVM exits. The JVM will only delete the temporary files when the JVM exits normally and obviously this isn't always the case.

> @TempDir annotation helped to replace createTempDirectory. But, i am afraid if there is any similar annotation like @TempDir as such for the createTempFile method.

Please review https://github.com/checkstyle/checkstyle/issues/14465#issuecomment-1938347487 and suggest....